velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +342,43 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
```

Comment:
The most succinct implementation could just inherit from Re2MatchConstantPattern and pass the pattern through to the constructor:
```
class LikeConstantPattern final : public Re2MatchConstantPattern<re2FullMatch> {
  explicit LikeConstantPattern(std:string pattern, char escapeChar) : Re2MatchConstantPattern(likePatternToRe2(pattern, escapeChar)) {}
```

however the error messages might be a little bit confusing / bad constant patterns would unconditionally result in an exception (probably not a huge deal).

Reply:
I'm going to pass on this. I feel the succinct-ness is saving few lines of code. The upside for making this change is not worth the downsides as much as they seem minimal.